fitbenchmarking/core/tests/test_results_fallback.py

```diff
@@ -0,0 +1,660 @@
+
+                _, sorted_results = preprocess_data(results)
+
+                all_sorted_results = [result for row in
+                                      sorted_results.values()
+                                      for result in row.values()]
+                for r in all_sorted_results:
+                    self.assertNotIn(None, r)
+
+
+class FallbackTagTests(unittest.TestCase):
+    """
+    Unit tests for finding fallback tags that
+    occur when there is a fallback on jacobian
+    and hessians.
+    """
+    def setUp(self):
+        """
+        Setting up the test results
+        """
+        # Loading the results from checkpoint.json
+        self.results, _ = load_mock_results()
+
+        self.sort_order = (['problem'],
+                           ['software',
+                            'minimizer',
+                            'jacobian',
+                            'hessian'])
+        self.col_sections = ['costfun']
+
+        # Defining the expected values for results in checkpoint.json
+        self.expected_tags = [{'row': 'prob_0', 'col': 's1:m10:j0:',
+                               'cat': 'cf1', 'result_ix': 0},
+                              {'row': 'prob_0', 'col': 's1:m11:j0:',
+                               'cat': 'cf1', 'result_ix': 1},
+                              {'row': 'prob_0', 'col': 's0:m01:j0:',
+                               'cat': 'cf1', 'result_ix': 2},
+                              {'row': 'prob_0', 'col': 's1:m10:j1:',
+                               'cat': 'cf1', 'result_ix': 4},
+                              {'row': 'prob_0', 'col': 's1:m11:j1:',
+                               'cat': 'cf1', 'result_ix': 5},
+                              {'row': 'prob_0', 'col': 's0:m01:j1:',
+                               'cat': 'cf1', 'result_ix': 6},
+                              {'row': 'prob_1', 'col': 's1:m10:j0:',
+                               'cat': 'cf1', 'result_ix': 7},
+                              {'row': 'prob_1', 'col': 's1:m11:j0:',
+                               'cat': 'cf1', 'result_ix': 8},
+                              {'row': 'prob_1', 'col': 's0:m01:j0:',
+                               'cat': 'cf1', 'result_ix': 9},
+                              {'row': 'prob_1', 'col': 's0:m00:j0:',
+                               'cat': 'cf1', 'result_ix': 10},
+                              {'row': 'prob_1', 'col': 's1:m10:j1:',
+                               'cat': 'cf1', 'result_ix': 11},
+                              {'row': 'prob_1', 'col': 's1:m11:j1:',
+                               'cat': 'cf1', 'result_ix': 12},
+                              {'row': 'prob_1', 'col': 's0:m01:j1:',
+                               'cat': 'cf1', 'result_ix': 13},
+                              {'row': 'prob_1', 'col': 's0:m00:j1:',
+                               'cat': 'cf1', 'result_ix': 14}]
+        self.expected_repeating_tags = []
+
+    def test_get_all_result_tags(self):
```

Comment:
This too could benefit from some more unit tests which explain what functionality the test is trying to confirm.